plugin/h5pimport/src/ZipPackageImporter.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+// For licensing terms, see /license.txt
+
+namespace Chamilo\PluginBundle\H5pImport\H5pImporter;
+
+use Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Class ZipPackageImporter.
+ */
+class ZipPackageImporter extends H5pPackageImporter
+{
+    private const ALLOWED_FILES = [
+        'json',
+        'png',
+        'jpg',
+        'jpeg',
+        'gif',
+        'bmp',
+        'tif',
+        'tiff',
+        'svg',
+        'eot',
+        'ttf',
+        'woff',
+        'woff2',
+        'otf',
+        'webm',
+        'mp4',
+        'ogg',
+        'mp3',
+        'm4a',
+        'wav',
+        'txt',
+        'pdf',
+        'rtf',
+        'doc',
+        'docx',
+        'xls',
+        'xlsx',
+        'ppt',
+        'pptx',
+        'odt',
+        'ods',
+        'odp',
+        'xml',
+        'csv',
+        'diff',
+        'patch',
+        'swf',
+        'md',
+        'textile',
+        'vtt',
+        'webvtt',
+        'gltf',
+        'gl',
+        'js',
+        'css',
+    ];
+
+    public function import(): string
+    {
+        $zipFile = new \PclZip($this->packageFileInfo['tmp_name']);
+        $zipContent = $zipFile->listContent();
+
+        if ($this->validateH5pPackageContent($zipContent)) {
+            $packageSize = array_reduce(
+                $zipContent,
+                function ($accumulator, $zipEntry) {
+                    return $accumulator + $zipEntry['size'];
+                }
+            );
+
+            $this->validateEnoughSpace($packageSize);
+
+            $pathInfo = pathinfo($this->packageFileInfo['name']);
+
+            $packageDirectoryPath = $this->generatePackageDirectory($pathInfo['filename']);
+            $zipFile->extract($packageDirectoryPath);
+
+            return "{$packageDirectoryPath}";
+        }
+
+        throw new \Exception('Invalid H5P package');
+    }
+
+    /**
+     * @throws \Exception
+     */
+    protected function validateEnoughSpace(int $packageSize)
+    {
+        $courseSpaceQuota = \DocumentManager::get_course_quota($this->course->getCode());
+
+        if (!enough_size($packageSize, $this->courseDirectoryPath, $courseSpaceQuota)) {
+            throw new \Exception('Not enough space to store package.');
+        }
+    }
+
+    /**
+     * Validate an H5P package.
+     * Check if 'h5p.json' and 'content/content.json' files exist
```

Comment:
From the code, it looks like it's 'h5p.json' *or* 'content/content.json', not *and*.